test/metabase/query_processor/middleware/cache_backend/db_test.clj

```diff
@@ -0,0 +1,43 @@
+(ns metabase.query-processor.middleware.cache-backend.db-test
+  (:require
+   [buddy.core.codecs :as codecs]
+   [clojure.string :as str]
+   [clojure.test :refer :all]
+   [metabase.db :as mdb]
+   [metabase.query-processor.middleware.cache-backend.interface :as i]
+   [metabase.test :as mt]
+   [metabase.util.encryption-test :as encryption-test])
+  (:import (java.sql Connection ResultSet)
+           (org.apache.commons.io IOUtils)))
+
+(set! *warn-on-reflection* true)
+
+(defn- cache-results
+  "Get the stored value from the query_cache"
+  ^bytes [^Connection conn]
+  (with-open [stmt (.prepareStatement conn "select results from query_cache"
+                                      ResultSet/TYPE_FORWARD_ONLY
+                                      ResultSet/CONCUR_READ_ONLY
+                                      ResultSet/CLOSE_CURSORS_AT_COMMIT)]
+    (with-open [rs (.executeQuery stmt)]
+      (is (.next rs))
+      (IOUtils/toByteArray (.getBinaryStream rs 1)))))
```

Comment:
This is fine, but for testing purposes we don't need to stream results out of the DB, so I think a simple select would be fine as well